src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -3243,6 +3369,20 @@ public IEnumerator<TElement> GetEnumerator()
                 for (int i = 0; i < buffer.count; i++) yield return buffer.items[map[i]];
             }
         }
+        
+        public TElement[] ToArray()
+        {
+            Buffer<TElement> buffer = new Buffer<TElement>(source);
+            int count = buffer.count;
+            TElement[] array = new TElement[count];
+            if (count > 0)
+            {
+                int[] map = GetEnumerableSorter(null).Sort(buffer.items, count);
```

Comment:
It'd be nice if this paralleled the same construct as in GetEnumerator above.  Since this version is more concise, could you change the three corresponding lines in GetEnumerator to match this, e.g.
```C#
int[] map = GetEnumerableSorter(null).Sort(buffer.items, buffer.count);
```